Allow systemd_cryptsetup_generator_t to write kmsg

ca-hu opened the pull request:
Addresses:
```
type=AVC msg=audit(1721045161.675:334): avc:  denied  { write } for  pid=1662 comm="systemd-cryptse" name="kmsg" dev="devtmpfs" ino=10 scontext=system_u:system_r:systemd_cryptsetup_generator_t:s0 tcontext=system_u:object_r:kmsg_device_t:s0 tclass=chr_file permissive=0
```

Reproducer:
https://openqa.opensuse.org/tests/4341371#downloads
encrypt during install, then `systemctl daemon-reload`

it might be better to allow this for all generators? idk
see: https://www.freedesktop.org/software/systemd/man/latest/systemd.generator.html
`Since syslog(3) is not available (see above), log messages have to be written to /dev/kmsg instead.`

zpytela commented:
@ca-hu You are right, please apply the change to `systemd_generator`; note the interfaces order should be alphabetical, but kernel goes first.

ca-hu committed:
Allow systemd_generator to write kmsg

see: https://www.freedesktop.org/software/systemd/man/latest/systemd.generator.html
"Since syslog(3) is not available (see above), log messages have to be written to /dev/kmsg instead."

Addresses:
type=AVC msg=audit(1721045161.675:334): avc:  denied  { write } for  pid=1662 comm="systemd-cryptse" name="kmsg" dev="devtmpfs" ino=10 scontext=system_u:system_r:systemd_cryptsetup_generator_t:s0 tcontext=system_u:object_r:kmsg_device_t:s0 tclass=chr_file permissive=0

Signed-off-by: Cathy Hu <cahu@suse.de>

ca-hu commented:
hope i understand the order correctly, if not let me know :)

zpytela commented:
purrfect

zpytela commented:
Merging, thank you.
